libraries/AP_ADSB/AP_ADSB.cpp

```diff
@@ -166,51 +168,73 @@ const AP_Param::GroupInfo AP_ADSB::var_info[] = {
 AP_ADSB::AP_ADSB()
 {
     AP_Param::setup_object_defaults(this, var_info);
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
     if (_singleton != nullptr) {
         AP_HAL::panic("AP_ADSB must be singleton");
     }
+#endif
     _singleton = this;
+
+#ifdef ADSB_STATIC_CALLSIGN
+    memcpy(&out_state.cfg.callsign, ADSB_STATIC_CALLSIGN, 8);
+#endif
 }
 
 /*
  * Initialize variables and allocate memory for array
  */
 void AP_ADSB::init(void)
 {
-    // in_state
-    in_state.vehicle_count = 0;
     if (in_state.vehicle_list == nullptr) {
-        if (in_state.list_size_param != constrain_int16(in_state.list_size_param, 1, ADSB_VEHICLE_LIST_SIZE_MAX)) {
-            in_state.list_size_param.set_and_notify(ADSB_VEHICLE_LIST_SIZE_DEFAULT);
-            in_state.list_size_param.save();
-        }
-        in_state.list_size = in_state.list_size_param;
-        in_state.vehicle_list = new adsb_vehicle_t[in_state.list_size];
+        // sanity check param
+        in_state.list_size_param = constrain_int16(in_state.list_size_param, 1, ADSB_VEHICLE_LIST_SIZE_MAX);
```

Comment:
This bit is a little tricky.  We can only store signed numbers in `AP_Param`, but a list size must be non-negative.  So this constrain isn't right.  It is pre-existing, 'though, so *shrug*.

Reply:
I don't understand. It's constraining from 1 to 100. Both are positive. note, it's not allowing 0 as a list size.

Reply:
It should be `constrain_uint16`.  It doesn't really matter.

Reply:
oh I see. But I'm then assigning it to an int16_t so then it's re-casting anyway.. so I think it's actually better this way.

Reply:
It might *look* like an int16_t, but it is interpreted as a uint by the code.  ATM if `ADSB_VEHICLE_LIST_SIZE_MAX` was 40,000 your `constrain_int16` call would be non-sensical as the latter value would be out-of-bound for a signed 16-bit integer.  It is entirely within range of the *intent* of the parameter, an unsigned integer.  The code is rife with places where we treat AP_Int* as unsigned.